src/client/java/teammates/client/scripts/DataMigrationForOrphanedStudentProfile.java

```diff
@@ -0,0 +1,45 @@
+package teammates.client.scripts;
+
+import java.io.IOException;
+
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.cmd.Query;
+
+import teammates.storage.entity.StudentProfile;
+
+/**
+ * Script to delete all orphaned StudentProfile entity.
+ */
+public class DataMigrationForOrphanedStudentProfile extends DataMigrationWithCheckpointForEntities<StudentProfile> {
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForOrphanedStudentProfile().doOperationRemotely();
+    }
+
+    @Override
+    protected Query<StudentProfile> getFilterQuery() {
+        return ofy().load().type(StudentProfile.class);
```

Comment:
Should this be the default `getFilterQuery`?

Reply:
This method is an abstract method the framework is designed to be used in other migration script (such as `DataMigrationForSessions`, we don't have a chance to run it as the data size is too big).

Reply:
I mean using the generic T instead of `StudentProfile`but then there's the problem of figuring out the runtime class of T to pass to this function

Reply:
And we may want to have addition filtering instead of just loading of all entities.

e.g.
```
ofy()
    .load()
    .type(Account.class)
    .filter('isInsturctor =', true)
```

although still doable if we want to auto-detect the generic class.

The [detection](https://stackoverflow.com/questions/1004022/java-generic-class-determine-type) is doable although but will introduce unnecessary complication I think.